presto-main/src/main/java/io/prestosql/sql/planner/LocalDynamicFilterConsumer.java

```diff
@@ -38,37 +39,41 @@
 import static com.google.common.base.MoreObjects.toStringHelper;
 import static com.google.common.base.Verify.verify;
 import static com.google.common.collect.ImmutableMap.toImmutableMap;
+import static com.google.common.util.concurrent.MoreExecutors.directExecutor;
 import static io.prestosql.sql.DynamicFilters.Descriptor;
 import static io.prestosql.sql.DynamicFilters.extractDynamicFilters;
 import static java.util.Objects.requireNonNull;
 import static java.util.function.Function.identity;
 
-public class LocalDynamicFilter
+public class LocalDynamicFilterConsumer
 {
-    private static final Logger log = Logger.get(LocalDynamicFilter.class);
+    private static final Logger log = Logger.get(LocalDynamicFilterConsumer.class);
 
     // Mapping from dynamic filter ID to its probe symbols.
     private final Multimap<String, Symbol> probeSymbols;
 
     // Mapping from dynamic filter ID to its build channel indices.
     private final Map<String, Integer> buildChannels;
 
-    private final TypeProvider types;
+    // Mapping from dynamic filter ID to its build channel type.
+    private final Map<String, Type> filterBuildTypes;
 
-    private final SettableFuture<Map<Symbol, Domain>> resultFuture;
+    private final SettableFuture<TupleDomain<String>> resultFuture;
```

Comment:
I shouldn't break that since we have one TupleDomain consumer per join node. Also, here I've only rearranged the code to convert from TupleDomain<String> to Map<String, Domain> or Map<Symbol, Domain> in getDynamicFilterDomains and getNodeLocalDynamicFilterForSymbols respectively, instead of converting only for node local DF in addPartition.